src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/ActivityFunction/FunctionReturnTypeAnalyzer.cs

```diff
@@ -49,20 +50,23 @@ public class FunctionReturnTypeAnalyzer
 
         private static bool InputMatchesOrTaskOrCompatibleType(ITypeSymbol invocationReturnType, ITypeSymbol definitionReturnType)
```

Comment:
This name is getting a bit hard to parse.

I would consider something like `IsReturnTypeValidForDefinition()`, and have a brief comment covering all of the cases (i.e. direct type match, compatible type match, or Task containing the matching type).

Reply:
Very good point